src/calc/calculate_phase_final.py

```diff
@@ -29,7 +29,19 @@ def calculate(self, reward_data5, total_amount):
                     rounding=ROUND_HALF_DOWN
                 )
             )
-            rl.payable = rl.type in [TYPE_FOUNDER, TYPE_OWNER, TYPE_DELEGATOR]
+            if adjustments and rl.address in adjustments.keys():
+                rl.adjustment = max(-adjustments[rl.address], -rl.amount)
+            else:
+                rl.adjustment = 0
+            rl.adjusted_amount = int(
+                Decimal((rl.ratio * total_amount) + rl.adjustment).to_integral_value(
+                    rounding=ROUND_HALF_DOWN
+                )
+            )
+            rl.payable = (
```

Comment:
the condition adjusted_amount > 0 is this correct for -5? sounds like this would be at 0 and it would skip and it should not (compared to the original logic)

Reply:
We are using this formula: `adjusted_amount = amount + adjustment`

In case of `-5`, `rl.adjustment` would be zero so `rl.adjusted_amount` is equal to `rl.amount`. So it's only marked payable if it's positive.

Reply:
Thanks for the precision

Reply:
Thanks for the precision